run_evaluation.py

```diff
@@ -90,4 +90,77 @@
       y.append(keypoints[i,1])
 plt.scatter(x,y)
 plt.imshow(img)
+
+# %% Run Predictions
+import imp
+
+import hourglass
+imp.reload(hourglass)
+from hourglass import HourglassNet
+
+import data_generator
+imp.reload(data_generator)
+from data_generator import DataGenerator
+
+import evaluation
+imp.reload(evaluation)
+from evaluation import Evaluation
+
+from constants import *
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+from pycocotools.coco import COCO
+from pycocotools.cocoeval import COCOeval
+
+# h = HourglassNet(NUM_COCO_KEYPOINTS,DEFAULT_NUM_HG,INPUT_CHANNELS,INPUT_DIM,OUTPUT_DIM)
+representative_set_df = pd.read_pickle(os.path.join(DEFAULT_PICKLE_PATH, 'representative_set.pkl'))
+# train_df, val_df = h.load_and_filter_annotations(DEFAULT_TRAIN_ANNOT_PATH,DEFAULT_VAL_ANNOT_PATH,0.1)
+
+subdir = '2021-03-29-11h-07m_batchsize_16_hg_4_loss_weighted_mse_aug_light_sigma4_learningrate_5.0e-03_opt_rmsProp_gt-4kp_activ_linear_subset_0.50_wmse-5'
+eval = Evaluation(
+    # ensure model_json and weights files exist in current directory and num_hg_blocks matches model_json
+    model_sub_dir = subdir
+    ,epoch = 15)
+print("Created Evaluation instance")
+
+generator = DataGenerator(
+            df=representative_set_df,
+            base_dir=DEFAULT_VAL_IMG_PATH,
+            input_dim=INPUT_DIM,
+            output_dim=OUTPUT_DIM,
+            num_hg_blocks=DEFAULT_NUM_HG,
+            shuffle=False,
+            batch_size=2,
+            online_fetch=True,
+            is_eval=True)
+print("Created DataGen instance")
+
+image_ids = eval.predict_keypoints(generator, location='output/predictions.json')
+print('Doneee')
+
+
+# %%
+annFile = "person_keypoints_val2017.json"
+cocoGt=COCO(annFile)
+
+resFile = "output/predictions.json"
```

Comment:
we'll need to save these in a unique path. (I'd recommend taking a look at Julian's PR #16  to see how he saves output files

Reply:
I think `run_evaluation.py` is just an example of how to use the methods introduced in this PR. I think it may be better suited to add a function that automates saving files with unique names in a second smaller PR given the time constraints. What do you think?

Reply:
Yep, sounds like a good plan.